sys/cddl/dev/dtrace/mips/dtrace_isa.c

```diff
@@ -151,10 +151,10 @@ dtrace_getupcstack(uint64_t *pcstack, int pcstack_limit)
 	if (pcstack_limit <= 0)
 		return;
 
-	pc = (uint64_t)tf->pc;
-	sp = (uint64_t)tf->sp;
-	ra = (uint64_t)tf->ra;
-	*pcstack++ = (uint64_t)tf->pc;
+	pc = TRAPF_PC(tf);
+	sp = (uintcap_t)tf->sp;
```

Comment:
Not sure what this function is supposed to do, just remember that that the stack pointer and link register in purecap are c11 and c17.

Reply:
right, this code doesn't work for Cheri. I had to put TRAPF_PC only to make it compile